packages/core/src/Utils/Analytics/analytics.ts

```diff
@@ -1,48 +1,184 @@
-interface AnalyticsEvent {
-    name: string;
-    properties: {
-        [key: string]: string;
-    };
+import { Analytics } from '@deriv-com/analytics';
+
+interface Payload {
+    type: string;
+    anonymousId: string;
 }
 
-const handleCachedEvents = () => {
-    let eventQueue: AnalyticsEvent[] = [];
-    const storedEvents = localStorage.getItem('pending_events');
-    try {
-        if (storedEvents) {
-            eventQueue = JSON.parse(storedEvents) as AnalyticsEvent[];
-            if (eventQueue.length > 0) {
-                eventQueue.forEach(event => {
-                    window.rudderanalytics.track(event.name, event.properties);
+type ResponseData = {
+    url: string;
+    method: string;
+    status: number;
+    headers: string;
+    data: string;
+    payload: Payload;
+};
+type Event = {
+    name: string;
+    properties: Record<string, string>;
+    cache?: boolean;
+};
+type Item = {
+    event: Event;
+    cache?: boolean;
+    callback?: (e: Event) => Event;
+};
+const cacheTrackEvents = {
+    interval: null as NodeJS.Timeout | null,
+    responses: [] as ResponseData[],
+    isTrackingResponses: false,
+    hash: (inputString: string, desiredLength = 32): string => {
+        const fnv1aHash = (string: string): number => {
+            let hash = 0x811c9dc5;
+            for (let i = 0; i < string.length; i++) {
```

Comment:
email hashing is not needed..you can remove it

Reply:
Fixed.